src/base_client.py

```
@@ -0,0 +1,102 @@
+        
+        Args:
+            base_url (str): Base URL for the API
+            timeout (int, optional): Request timeout in seconds. Defaults to 10.
+            log_level (int, optional): Logging level. Defaults to logging.INFO.
+        """
+        self.base_url = base_url
+        self.timeout = timeout
+        
+        # Configure logging
+        logging.basicConfig(
+            level=log_level, 
```

Pylint (3.3.7)

[convention] Trailing whitespace

(C0303)